### DISCUSSION TITLE
Should we relax our linting?

### DISCUSSION THREAD

**Discussion by WebReflection** [pyscript/pyscript#1523] [General]
*TL;DR* if peers reading code are not happy about the code I think we're making reviews harder for no real-world benefits: see [this comment](https://github.com/pyscript/pyscript/pull/1522#discussion_r1226682626)

- - -

I am all for "same-style" approach to code in general but on the other hand, I am not in favor of strongly opinionated style that result into friction over well intended code from developers. The *next* branch is to me now unreadable, and I wrote that code, compared to my initial code intents, and it's harder to follow, harder to explain, group, reason about, or understand.

All changes have been applied by our current pre-commit hooks and while when I read TS after prettier the amount of noise in the code is superior to the amount of fixes that prettier is trying to impose, for plain JS with well defined TS comments things start looking a bit awkward.

The code is still 100% TS friendly and it will show a TS logo on npm whenever we publish, but internals are simpler to read and follow ... in short: do we have any reason to keep prettier this strict and opinionated for stuff that would otherwise be just easier to contribute to? (to me and to anyone reading my code *after* prettier).

Thanks for considering some relaxing if you find it convenient as I do!

**Comment by WebReflection** [comment]
P.S. this is the least problem we have and there's zero urgency in reacting, but I've thought after both me and Fabio were a bit disappointed by the code style imposed by the tool maybe in the future we can think about relaxing here and there some part of the defaults/rules we have. I am (at the end of the day) OK in prettier fixing/changing code automatically, as long as it still works as intended :+1:

**Comment by WebReflection** [comment]
Big face-palm here ... it looks like the rules we use for PyScript are already better, so I need to investigate what is causing my folder to get different rules we had already in PyScript ... it might be just a matter of extensions but we can happily ignore this conversation until I find the culprit of the current prettier story within the `pyscript.core` folder.
